earth_enterprise/SConstruct

```diff
@@ -26,6 +26,8 @@ EnsureSConsVersion(0, 96, 92)
 
 # process commandline arguments
 installdir = ARGUMENTS.get('installdir', '/tmp/fusion_os_install')
+with open('rpms/build.properties', 'w') as f:
+    print >> f, str('prefix=') + installdir.strip()
```

Comment:
Is there any mechanism that would avoid adding this over and over if you build more than once without cleaning?

Reply:
Well, you get it every time you run scons, because you may change installdir in any run.  I also wanted to push the version info and maybe other things thru properties later, which also may change.

Reply:
Right, my point is that you are only appending this string to the file rather than recreating the file from scratch each time, so you'll end up with something like:
```
prefix=/tmp/install_os_install
prefix=/some/other/path
```
I'm not sure what gradle will do when it reads that. So, I'd either delete the file every time, but then it would be good to add a message that says it's autogenerated and shouldn't be edited by hand. Or, actually replace the line with prefix rather than append.

Reply:
Actually, it opens it for 'w' at the beginning, so it never appends to existing file, it rewrites, so there is always only one prefix= line.  Append would make sense for adding more vars though once it is open write....

Reply:
Ah, I am not fluent in the details of Python behavior it seems. After consulting the documentation, I see that it will truncate the file. But, you're right about it being a little awkward if you wanted to add more values later. But, as long as it's within the "with" block you should be fine. I'd still suggest adding a comment stating it's an auto-generated file though. If someone saw that sitting in one of the directories they may think it's a valid place to set properties.